src/NServiceBus.Core/Routing/MessageDrivenSubscriptions/Publishers.cs

```diff
@@ -62,11 +62,11 @@ static PublisherAddress StaticTypeRule(Type typeBeingQueried, Type configuredTyp
         }
 
         /// <summary>
-        /// Adds a dynamic rule.
+        /// Adds a dynamic rule which is invoked on each subscription to determine the address of the publisher.
         /// </summary>
-        /// <param name="dynamicRule">The rule.</param>
-        /// <param name="description">Optional description of the rule.</param>
-        public void AddDynamic(Func<Type, PublisherAddress> dynamicRule, string description = null)
+        /// <param name="dynamicRule">The rule returning an address or <c>null</c> for a type.</param>
+        /// <param name="description">A description of the rule.</param>
+        public void Add(Func<Type, PublisherAddress> dynamicRule, string description)
```

Comment:
I guess dynamic isn't a very specific description or it doesn't make very clear that this rule invoked every time so it would probably be best to rename the parameter to. No strong opinions though. Maybe @weralabaj wants to chime in since she raised the original comment about that API.